crates/emitter/src/forward_jump_emitter.rs

```diff
@@ -0,0 +1,73 @@
+use super::emitter::{BytecodeOffset, EmitError};
+use crate::ast_emitter::AstEmitter;
+
+#[derive(Debug)]
+pub enum JumpKind {
+    Coalesce,
+    LogicalAnd,
+    LogicalOr,
+    IfEq,
+    Goto,
+}
+
+#[derive(Debug)]
+#[must_use]
+pub struct JumpPatchEmitter {
+    pub offset: BytecodeOffset,
```

Comment:
this field should be private, given `JumpPatchEmitter` shouldn't be constructible outside of `ForwardJumpEmitter`.